test/e2e/policyrecommendation_test.go

```diff
@@ -167,6 +167,54 @@ func TestPolicyRecommendation(t *testing.T) {
 			testPolicyRecommendationRetrieve(t, data, true, testFlowPodToPod, testFlowPodToSvc, testFlowPodToExternal)
 		})
 	}
+
+	t.Run("testTheiaManagerRestart", func(t *testing.T) {
+		testTheiaManagerRestart(t, data)
+	})
+}
+
+func testTheiaManagerRestart(t *testing.T, data *TestData) {
+	_, jobName1, err := runJob(t, data)
+	require.NoError(t, err)
+	_, jobName2, err := runJob(t, data)
+	require.NoError(t, err)
+
+	// Simulate the Theia Manager downtime
+	cmd := "kubectl delete deployment theia-manager -n flow-visibility"
+	_, stdout, stderr, err := data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, fmt.Sprintf("error when running %s from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr))
+	// Delete the first job during the Theia Manager downtime
+	cmd = fmt.Sprintf("kubectl delete npr %s -n flow-visibility", jobName1)
+	_, stdout, stderr, err = data.RunCommandOnNode(controlPlaneNodeName(), cmd)
+	require.NoErrorf(t, err, fmt.Sprintf("error when running %s from %s: %v\nstdout:%s\nstderr:%s", cmd, controlPlaneNodeName(), err, stdout, stderr))
+	// Redeploy the Theia Manager
+	err = data.deployFlowVisibilityCommon(flowVisibilityWithSparkYML)
+	require.NoError(t, err)
+	// Sleep for a short period to wait the deployment
+	time.Sleep(1 * time.Second)
+	theiaManagerPodName, err := data.getPodByLabel(theiaManagerPodLabel, flowVisibilityNamespace)
+	require.NoError(t, err)
+	err = data.podWaitForReady(defaultTimeout, theiaManagerPodName, flowVisibilityNamespace)
+	require.NoError(t, err)
+
+	// Check the status of jobName2
+	stdout, err = getJobStatus(t, data, jobName2)
+	require.NoError(t, err)
+	assert := assert.New(t)
+	assert.Containsf(stdout, "Status of this policy recommendation job is", "stdout: %s", stdout)
+	err = data.podWaitForReady(defaultTimeout, jobName2+"-driver", flowVisibilityNamespace)
+	require.NoError(t, err)
+	_, err = deleteJob(t, data, jobName2)
+	require.NoError(t, err)
+
+	// Check the SparkApplication and database entries of jobName1 do not exist
+	cmd = fmt.Sprintf("kubectl get sparkapplication %s -n flow-visibility", jobName1)
```

Comment:
Maybe wraps this and and L215 in some retries just in case theia manager gets slow in removing them?

Reply:
Updated, thanks! Also add some retries for npr deletion as I found some failures.